diff --git a/src/pk-engine.c b/src/pk-engine.c
index 6b145b6..26e2ca7 100644
--- a/src/pk-engine.c
+++ b/src/pk-engine.c
@@ -51,11 +51,6 @@
 #include "pk-transaction.h"
 #include "pk-scheduler.h"
 
-#ifndef glib_autoptr_cleanup_PolkitAuthorizationResult
-G_DEFINE_AUTOPTR_CLEANUP_FUNC(PolkitAuthorizationResult, g_object_unref)
-G_DEFINE_AUTOPTR_CLEANUP_FUNC(PolkitSubject, g_object_unref)
-#endif
-
 static void     pk_engine_finalize	(GObject       *object);
 static void	pk_engine_set_locked (PkEngine *engine, gboolean is_locked);
 
diff --git a/src/pk-transaction.c b/src/pk-transaction.c
index de7953c..7a3c8dd 100644
--- a/src/pk-transaction.c
+++ b/src/pk-transaction.c
@@ -55,11 +55,6 @@
 #include "pk-transaction.h"
 #include "pk-transaction-private.h"
 
-#ifndef glib_autoptr_cleanup_PolkitAuthorizationResult
-G_DEFINE_AUTOPTR_CLEANUP_FUNC(PolkitAuthorizationResult, g_object_unref)
-G_DEFINE_AUTOPTR_CLEANUP_FUNC(PolkitDetails, g_object_unref)
-#endif
-
 static void     pk_transaction_finalize		(GObject	    *object);
 static void     pk_transaction_dispose		(GObject	    *object);
 
