diff --git a/backends/urpmi/pk-backend-urpmi.c b/backends/urpmi/pk-backend-urpmi.c
index 08e9fb9..040bb7a 100644
--- a/backends/urpmi/pk-backend-urpmi.c
+++ b/backends/urpmi/pk-backend-urpmi.c
@@ -66,12 +66,6 @@ pk_backend_initialize (PkBackend *backend)
 {
 	g_debug ("backend: initialize");
 
-	/* BACKEND MAINTAINER: feel free to remove this when you've
-	 * added support for ONLY_DOWNLOAD as specified in backends/PORTING.txt
-	 * */
-	g_error ("Backend needs to be ported to 0.8.x -- "
-		 "see backends/PORTING.txt for details");
-
 	spawn = pk_backend_spawn_new ();
 	pk_backend_spawn_set_name (spawn, "urpmi");
 }
