diff -Naur PackageKit-1.0.8/backends/urpmi/helpers/urpmi-dispatched-backend.pl PackageKit-1.0.8.tpg/backends/urpmi/helpers/urpmi-dispatched-backend.pl
--- PackageKit-1.0.8/backends/urpmi/helpers/urpmi-dispatched-backend.pl	2015-08-08 07:15:27.000000000 +0000
+++ PackageKit-1.0.8.tpg/backends/urpmi/helpers/urpmi-dispatched-backend.pl	2015-09-06 03:08:07.003344390 +0000
@@ -192,7 +192,16 @@
   pk_print_status(PK_STATUS_ENUM_QUERY);
 
   my %product_id = %{ urpm::mirrors::parse_LDAP_namespace_structure(cat_('/etc/product.id')) };
-  my @distribs = urpm::mirrors::_mirrors_filtered($urpm, urpm::mirrors::_mageia_mirrorlist(\%product_id));
+  my @distribs;
+
+  if (is_mageia()) {
+      push(@distribs, urpm::mirrors::_mirrors_filtered($urpm, urpm::mirrors::_mageia_mirrorlist(\%product_id)));
+  } elsif (is_omv()) {
+      push(@distribs, urpm::mirrors::_mirrors_filtered($urpm, urpm::mirrors::_mandriva_mirrorlist(\%product_id)));
+  } else {
+      pk_print_error(PK_ERROR_ENUM_TRANSACTION_ERROR, "Could not determine system setup.\n");
+      die;
+  }
 
   my ($distrib) = grep { $_->{version} == $product_id{version} } @distribs;
 
@@ -201,7 +210,14 @@
 
   my $newer_version = _get_newer_distrib($distrib->{version}, \@distribs);
   $newer_version or goto finished;
-  pk_print_distro_upgrade(PK_DISTRO_UPGRADE_ENUM_STABLE, join(" ", "Mageia", $product_id{product}, $newer_version->{version}), "");
+  if (is_mageia()) {
+      pk_print_distro_upgrade(PK_DISTRO_UPGRADE_ENUM_STABLE, join(" ", "Mageia", $product_id{product}, $newer_version->{version}), "");
+  } elsif (is_omv()) {
+      pk_print_distro_upgrade(PK_DISTRO_UPGRADE_ENUM_STABLE, join(" ", "OpenMandriva", $product_id{product}, $newer_version->{version}), "");
+  } else {
+      pk_print_error(PK_ERROR_ENUM_TRANSACTION_ERROR, "Could not determine system setup.\n");
+      die;
+  }
 
   finished:
   _finished();
@@ -797,13 +813,16 @@
   }
   
   my @to_upgrade_pkids = map { get_installed_fullname_pkid($_) || () } @to_install;
+
+  my $site_link = "http://bugs.mageia.org" if is_mageia();
+  my $site_link = "http://issues.openmandriva.org" if is_omv();
   
   pk_print_update_detail(get_package_id($pkg),
     join("&", @to_upgrade_pkids),
     join("&", map { fullname_to_package_id($_) } @to_remove),
-    "http://bugs.mageia.org",
-    "http://bugs.mageia.org",
-    "http://bugs.mageia.org",
+      $site_link,
+      $site_link,
+      $site_link,
     $restart ? PK_RESTART_ENUM_SYSTEM : PK_RESTART_ENUM_NONE,
     $desc);
 }
diff -Naur PackageKit-1.0.8/backends/urpmi/helpers/urpmi_backend/tools.pm PackageKit-1.0.8.tpg/backends/urpmi/helpers/urpmi_backend/tools.pm
--- PackageKit-1.0.8/backends/urpmi/helpers/urpmi_backend/tools.pm	2015-06-15 14:27:05.000000000 +0000
+++ PackageKit-1.0.8.tpg/backends/urpmi/helpers/urpmi_backend/tools.pm	2015-09-06 02:30:59.498435035 +0000
@@ -26,6 +26,7 @@
   urpm_name 
   find_installed_fullname 
   is_mageia
+  is_omv
   is_package_installed
   get_package_id 
   ensure_utf8 
@@ -94,8 +95,18 @@
     cat_('/etc/release') =~ /Mageia/;
 }
 
+sub is_omv() {
+    cat_('/etc/release') =~ /OpenMandriva/;
+}
+
 sub vendor() {
-    is_mageia() ? "mageia" : "mandriva";
+    if (is_mageia()) {
+        return "mageia";
+    } elsif (is_omv()) {
+        return "openmandriva";
+    } else {
+        return "mandriva";
+    }
 }
 
 sub get_package_id {
