diff -up PackageKit-1.1.12/backends/zypp/Makefile.am.omv~ PackageKit-1.1.12/backends/zypp/Makefile.am
--- PackageKit-1.1.12/backends/zypp/Makefile.am.omv~	2019-11-11 01:39:38.577139951 +0100
+++ PackageKit-1.1.12/backends/zypp/Makefile.am	2019-11-11 01:39:49.382183500 +0100
@@ -8,7 +8,7 @@ libpk_backend_zypp_la_SOURCES =	pk-backe
 libpk_backend_zypp_la_LIBADD = $(PK_PLUGIN_LIBS)
 libpk_backend_zypp_la_LDFLAGS = -module -avoid-version $(ZYPP_LIBS)
 libpk_backend_zypp_la_CFLAGS = $(PK_PLUGIN_CFLAGS) $(WARNINGFLAGS_CPP)
-libpk_backend_zypp_la_CXXFLAGS = $(PK_PLUGIN_CXXFLAGS) --std=c++0x -Wall -Woverloaded-virtual -Wnon-virtual-dtor
+libpk_backend_zypp_la_CXXFLAGS = $(PK_PLUGIN_CXXFLAGS) --std=c++17 -Wall -Woverloaded-virtual -Wnon-virtual-dtor
 libpk_backend_zypp_la_CPPFLAGS = $(PK_PLUGIN_CFLAGS) $(ZYPP_CFLAGS) -Wno-deprecated -D_FILE_OFFSET_BITS=64
 
 -include $(top_srcdir)/git.mk
