--- ./contrib/command-not-found/PackageKit.sh.in.orig	2011-05-25 11:36:03.000000000 +0200
+++ ./contrib/command-not-found/PackageKit.sh.in	2011-05-25 11:36:46.000000000 +0200
@@ -14,7 +14,7 @@
 	[ ! -S /var/run/dbus/system_bus_socket ] && runcnf=0
 
 	# don't run if packagekitd doesn't exist in the _system_ root
-	[ ! -x /usr/libexec/packagekitd ] && runcnf=0
+	[ ! -x @LIBEXECDIR@/packagekitd ] && runcnf=0
 
 	# run the command, or just print a warning
 	if [ $runcnf -eq 1 ]; then
--- ./contrib/command-not-found/PackageKit.sh.orig	2011-05-25 11:40:25.000000000 +0200
+++ ./contrib/command-not-found/PackageKit.sh	2011-05-25 11:40:38.000000000 +0200
@@ -14,7 +14,7 @@
 	[ ! -S /var/run/dbus/system_bus_socket ] && runcnf=0
 
 	# don't run if packagekitd doesn't exist in the _system_ root
-	[ ! -x /usr/libexec/packagekitd ] && runcnf=0
+	[ ! -x /usr/lib/packagekitd ] && runcnf=0
 
 	# run the command, or just print a warning
 	if [ $runcnf -eq 1 ]; then
