--- ./etc/Vendor.conf~	2008-10-05 19:35:08.000000000 +0200
+++ ./etc/Vendor.conf	2008-10-12 15:22:23.000000000 +0200
@@ -12,7 +12,7 @@
 # If the value is set to 'none' then no link is shown.
 #
 # default=http://www.packagekit.org/pk-package-not-found.html
-DefaultUrl=http://www.packagekit.org/pk-package-not-found.html
+DefaultUrl=https://wwww.openmandriva.org/
 
 # The URL which is shown to the user when a codec could not be found.
 # It should explain why certain codecs cannot be used, and perhaps show
