diff -up PackageKit-1.1.12/backends/dnf/dnf-backend-vendor-openmandriva.c.omv~ PackageKit-1.1.12/backends/dnf/dnf-backend-vendor-openmandriva.c
--- PackageKit-1.1.12/backends/dnf/dnf-backend-vendor-openmandriva.c.omv~	2019-07-31 21:43:21.823723147 +0200
+++ PackageKit-1.1.12/backends/dnf/dnf-backend-vendor-openmandriva.c	2019-07-31 21:44:35.647519510 +0200
@@ -29,7 +29,7 @@ dnf_validate_supported_repo (const gchar
 	guint i, j, k, l;
 
 	const gchar *valid_sourcesect[] = { "",
-					  "-contrib",
+					  "-unsupported",
 					  "-restricted",
 					  "-non-free",
 					  NULL };
@@ -39,16 +39,22 @@ dnf_validate_supported_repo (const gchar
 					  "-source",
 					  NULL };
 
-	const gchar *valid_arch[] = { "x86_64",
+	const gchar *valid_arch[] = { "znver1",
+				      "x86_64",
 				      "i686",
 				      "aarch64",
-				      "armv7hl",
+				      "armv7hnl",
+				      "noarch",
 				      NULL };
 
 	const gchar *valid[] = { "openmandriva",
 				 "updates",
 				 "testing",
 				 "cooker",
+				 "rolling",
+				 "rock",
+				 "4.1",
+				 "4.0",
 				 NULL };
 
 	/* Iterate over the ID arrays to find a matching identifier */
