--- PackageKit-0.8.11/backends/urpmi/helpers/urpmi-dispatched-backend.pl.orig	2013-10-15 11:01:04.367796949 +0100
+++ PackageKit-0.8.11/backends/urpmi/helpers/urpmi-dispatched-backend.pl	2013-10-15 11:01:28.705797934 +0100
@@ -43,7 +43,7 @@
 dispatch_command($urpm, \@ARGV);
 print "finished\n";
 
-foreach (<STDIN>) {
+while (<STDIN>) {
   chomp($_);
   my @args = split(/\t/, $_);
   dispatch_command($urpm, \@args);
--- PackageKit-0.8.11/backends/urpmi/helpers/urpmi_backend/tools.pm.orig	2013-10-15 11:01:43.819798546 +0100
+++ PackageKit-0.8.11/backends/urpmi/helpers/urpmi_backend/tools.pm	2013-10-15 11:02:45.792801054 +0100
@@ -87,15 +87,15 @@
 
 sub is_package_installed {
     my ($pkg) = @_;
-    return URPM::DB::open()->is_package_installed($pkg);
+    return URPM::is_package_installed(URPM::DB::open(),$pkg);
 }
 
 sub is_mageia() {
-    cat_('/etc/release') =~ /Mageia/;
+    cat_('/etc/release') =~ /OpenMandriva/;
 }
 
 sub vendor() {
-    is_mageia() ? "mageia" : "mandriva";
+    is_mageia() ? "openmandriva" : "mandriva";
 }
 
 sub get_package_id {
