diff -up PackageKit-1.1.9/backends/dnf/dnf-backend-vendor-openmandriva.c.omv~ PackageKit-1.1.9/backends/dnf/dnf-backend-vendor-openmandriva.c
--- PackageKit-1.1.9/backends/dnf/dnf-backend-vendor-openmandriva.c.omv~	2018-04-03 12:13:17.828584294 +0200
+++ PackageKit-1.1.9/backends/dnf/dnf-backend-vendor-openmandriva.c	2018-04-03 12:15:38.293045882 +0200
@@ -0,0 +1,68 @@
+/* -*- Mode: C; tab-width: 8; indent-tabs-mode: t; c-basic-offset: 8 -*-
+ *
+ * Copyright (C) 2018 Bernhard Rosenkraenzer <bero@lindev.ch>
+ * based on dnf-backend-vendor-mageia.c
+ * Copyright (C) 2016 Neal Gompa <ngompa13@gmail.com>
+ *
+ * Licensed under the GNU Lesser General Public License Version 2.1
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301 USA
+ */
+
+#include "dnf-backend-vendor.h"
+
+gboolean
+dnf_validate_supported_repo (const gchar *id)
+{
+	guint i, j, k, l;
+
+	const gchar *valid_sourcesect[] = { "",
+					  "-contrib",
+					  "-restricted",
+					  "-non-free",
+					  NULL };
+
+	const gchar *valid_sourcetype[] = { "",
+					  "-debuginfo",
+					  "-source",
+					  NULL };
+
+	const gchar *valid_arch[] = { "x86_64",
+				      "i686",
+				      "aarch64",
+				      "armv7hl",
+				      NULL };
+
+	const gchar *valid[] = { "openmandriva",
+				 "updates",
+				 "testing",
+				 "cooker",
+				 NULL };
+
+	/* Iterate over the ID arrays to find a matching identifier */
+	for (i = 0; valid[i] != NULL; i++) {
+		for (j = 0; valid_arch[j] != NULL; j++) {
+			for (k = 0; valid_sourcesect[k] != NULL; k++) {
+				for (l = 0; valid_sourcetype[l] != NULL; l++) {
+					g_autofree gchar *source_entry = g_strconcat(valid[i], "-", valid_arch[j], valid_sourcesect[k], valid_sourcetype[l], NULL);
+					if (g_strcmp0 (id, source_entry) == 0) {
+						return TRUE;
+					}
+				}
+			}
+		}
+	}
+	return FALSE;
+}
diff -up PackageKit-1.1.9/backends/dnf/Makefile.am.omv~ PackageKit-1.1.9/backends/dnf/Makefile.am
--- PackageKit-1.1.9/backends/dnf/Makefile.am.omv~	2018-04-03 12:12:40.310460333 +0200
+++ PackageKit-1.1.9/backends/dnf/Makefile.am	2018-04-03 12:12:58.700521133 +0200
@@ -2,7 +2,8 @@ plugindir = $(PK_PLUGIN_DIR)
 plugin_LTLIBRARIES = libpk_backend_dnf.la
 EXTRA_DIST =								\
 	dnf-backend-vendor-fedora.c					\
-	dnf-backend-vendor-mageia.c
+	dnf-backend-vendor-mageia.c					\
+	dnf-backend-vendor-openmandriva.c
 libpk_backend_dnf_la_SOURCES =						\
 	dnf-backend-vendor.h						\
 	dnf-backend.c							\
diff -up PackageKit-1.1.9/configure.ac.omv~ PackageKit-1.1.9/configure.ac
--- PackageKit-1.1.9/configure.ac.omv~	2018-04-03 12:15:51.736089873 +0200
+++ PackageKit-1.1.9/configure.ac	2018-04-03 12:16:12.101156463 +0200
@@ -394,8 +394,8 @@ AC_SUBST(DBUS_SERVICES_DIR)
 if test x$enable_dnf = xyes; then
 	PKG_CHECK_MODULES(DNF, appstream-glib libdnf >= 0.7.0)
 	AC_ARG_WITH(dnf-vendor,
-			[AS_HELP_STRING([--with-dnf-vendor=<vendor>],[select a vendor configuration (fedora, mageia; default is fedora)])])
-	if test "$with_dnf_vendor" = "fedora" -o "$with_dnf_vendor" = "mageia"; then
+			[AS_HELP_STRING([--with-dnf-vendor=<vendor>],[select a vendor configuration (fedora, mageia, openmandriva; default is fedora)])])
+	if test "$with_dnf_vendor" = "fedora" -o "$with_dnf_vendor" = "mageia" -o "$with_dnf_vendor" = "openmandriva"; then
 		with_dnf_vendor="$with_dnf_vendor"
 	else
 		with_dnf_vendor="fedora"
