diff -Naur PackageKit-1.0.8/backends/urpmi/helpers/urpmi-dispatched-backend.pl PackageKit-1.0.8.tpg/backends/urpmi/helpers/urpmi-dispatched-backend.pl
--- PackageKit-1.0.8/backends/urpmi/helpers/urpmi-dispatched-backend.pl	2015-08-08 07:15:27.000000000 +0000
+++ PackageKit-1.0.8.tpg/backends/urpmi/helpers/urpmi-dispatched-backend.pl	2015-09-05 22:24:32.763139633 +0000
@@ -192,7 +192,11 @@
   pk_print_status(PK_STATUS_ENUM_QUERY);
 
   my %product_id = %{ urpm::mirrors::parse_LDAP_namespace_structure(cat_('/etc/product.id')) };
-  my @distribs = urpm::mirrors::_mirrors_filtered($urpm, urpm::mirrors::_mageia_mirrorlist(\%product_id));
+  if (is_mageia()) {
+   my @distribs = urpm::mirrors::_mirrors_filtered($urpm, urpm::mirrors::_mageia_mirrorlist(\%product_id));
+  } else {
+   my @distribs = urpm::mirrors::_mirrors_filtered($urpm, urpm::mirrors::_mandriva_mirrorlist(\%product_id));
+  }
 
   my ($distrib) = grep { $_->{version} == $product_id{version} } @distribs;
 
@@ -201,7 +205,13 @@
 
   my $newer_version = _get_newer_distrib($distrib->{version}, \@distribs);
   $newer_version or goto finished;
-  pk_print_distro_upgrade(PK_DISTRO_UPGRADE_ENUM_STABLE, join(" ", "Mageia", $product_id{product}, $newer_version->{version}), "");
+  if (is_mageia()) {
+    pk_print_distro_upgrade(PK_DISTRO_UPGRADE_ENUM_STABLE, join(" ", "Mageia", $product_id{product}, $newer_version->{version}), "");
+  } elsif (is_omv()) {
+    pk_print_distro_upgrade(PK_DISTRO_UPGRADE_ENUM_STABLE, join(" ", "OpenMandriva", $product_id{product}, $newer_version->{version}), "");
+  else {
+    print("Cannot determine your system");
+  }
 
   finished:
   _finished();
@@ -801,9 +811,16 @@
   pk_print_update_detail(get_package_id($pkg),
     join("&", @to_upgrade_pkids),
     join("&", map { fullname_to_package_id($_) } @to_remove),
-    "http://bugs.mageia.org",
-    "http://bugs.mageia.org",
-    "http://bugs.mageia.org",
+    if (is_mageia()) {
+      "http://bugs.mageia.org",
+      "http://bugs.mageia.org",
+      "http://bugs.mageia.org",
+    } elsif (is_omv()) {
+      "http://isuess.openmandriva.org",
+      "http://isuess.openmandriva.org",
+      "http://isuess.openmandriva.org",
+    }
+    else {}
     $restart ? PK_RESTART_ENUM_SYSTEM : PK_RESTART_ENUM_NONE,
     $desc);
 }
diff -Naur PackageKit-1.0.8/backends/urpmi/helpers/urpmi_backend/tools.pm PackageKit-1.0.8.tpg/backends/urpmi/helpers/urpmi_backend/tools.pm
--- PackageKit-1.0.8/backends/urpmi/helpers/urpmi_backend/tools.pm	2015-06-15 14:27:05.000000000 +0000
+++ PackageKit-1.0.8.tpg/backends/urpmi/helpers/urpmi_backend/tools.pm	2015-09-05 19:37:51.073709225 +0000
@@ -26,6 +26,7 @@
   urpm_name 
   find_installed_fullname 
   is_mageia
+  is_omv
   is_package_installed
   get_package_id 
   ensure_utf8 
@@ -94,8 +95,14 @@
     cat_('/etc/release') =~ /Mageia/;
 }
 
+sub is_omv() {
+    cat_('/etc/release') =~ /OpenMandriva/;
+}
+
+
 sub vendor() {
     is_mageia() ? "mageia" : "mandriva";
+    is_omv() ? "openmandriva" : "mandriva";
 }
 
 sub get_package_id {
